sycl/source/detail/event_impl.cpp

```diff
@@ -611,6 +611,99 @@ bool event_impl::isCompleted() {
 
 void event_impl::setCommand(Command *Cmd) { MCommand = Cmd; }
 
+template <bool LockQueue>
+void registerEventDependency(
+    const detail::EventImplPtr &EventImpl,
+    std::vector<detail::EventImplPtr> &EventsRegistered,
+    detail::queue_impl *QueueImpl, const detail::context_impl &ContextImpl,
+    const detail::device_impl &DeviceImpl,
+    const std::shared_ptr<ext::oneapi::experimental::detail::graph_impl> &Graph,
```

Comment:
We do not need `shared_ptr` here. We can just pass a raw pointer to the `graph_impl`, right?